garden-backend-service/src/api/routes/users.py

```diff
@@ -62,71 +63,124 @@ async def update_user_info(
+
+    if user is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"No user with user_uuid {user_uuid} found.",
+        )
+
     if garden is None:
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
             detail=f"No garden with doi {doi} found.",
         )
 
-    stmt = insert(users_saved_gardens).values(user_id=user.id, garden_id=garden.id)
+    await db.refresh(user, ["saved_gardens"])
+    for saved_garden in user.saved_gardens:
+        if garden.doi == saved_garden.doi:
+            raise HTTPException(
+                status_code=status.HTTP_409_CONFLICT,
+                detail=f"User has already saved garden with doi {garden.doi}",
+            )
+
+    user.saved_gardens.append(garden)
     try:
-        await db.execute(stmt)
         await db.commit()
-        return await _get_saved_garden_dois(authed_user, db)
+        return [
+            GardenMetadataResponse.from_orm(garden) for garden in user.saved_gardens
+        ]
     except IntegrityError as e:
         await db.rollback()
         raise HTTPException(
-            status_code=status.HTTP_400_BAD_REQUEST,
+            status_code=status.HTTP_409_CONFLICT,
             detail="User has already saved this garden.",
         ) from e
 
 
-@router.delete("/gardens/saved/{doi:path}")
+@router.delete("/{user_uuid}/saved/gardens/{doi:path}")
 async def remove_saved_garden(
+    user_uuid: UUID,
     doi: str,
     authed_user: User = Depends(authed_user),
     db: AsyncSession = Depends(get_db_session),
-) -> list[str]:
-    """Remove a garden from the authed users list of saved gardens by doi."""
-    user: User = await User.get(db, identity_id=authed_user.identity_id)
+) -> list[GardenMetadataResponse]:
+    """Remove a garden from the user's list of saved gardens by doi."""
+
+    if user_uuid != authed_user.identity_id:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail=f"Not authorized to remove a saved garden for user {user_uuid}.",
+        )
+
+    user: User | None = await User.get(db, identity_id=user_uuid)
     garden: Garden | None = await Garden.get(db, doi=doi)
 
+    if user is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"No user with user_uuid {user_uuid} found.",
+        )
+
     if garden is None:
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
             detail=f"No garden with doi {doi} found.",
         )
 
-    stmt = delete(users_saved_gardens).where(
-        and_(
-            users_saved_gardens.c.user_id == user.id,
-            users_saved_gardens.c.garden_id == garden.id,
+    await db.refresh(user, ["saved_gardens"])
+    updated_saved_gardens = list(filter(lambda g: g != garden, user.saved_gardens))
+    if len(updated_saved_gardens) == len(user.saved_gardens):
+        raise HTTPException(
```

Comment:
similar to the PUT method, DELETEs should be idempotent too -- I think we probably just want something totally naive here like 
```python
await db.refresh(user, ["saved_gardens"])
if garden in user.saved_gardens:
    user.saved_gardens.remove(garden)
```
without throwing an error if it's not saved / already deleted